src/coreclr/Directory.Build.props

```diff
@@ -1,28 +1,168 @@
+    <ConfigurationGroup Condition="'$(ConfigurationGroup)' == '' and $(Configuration.EndsWith('Checked'))">Checked</ConfigurationGroup>
+    <ConfigurationGroup Condition="'$(ConfigurationGroup)' == ''">Debug</ConfigurationGroup>
+
+    <OSGroup Condition="$(Configuration.StartsWith('Windows'))">Windows_NT</OSGroup>
+    <OSGroup Condition="$(Configuration.StartsWith('Unix'))">Unix</OSGroup>
+    <OSGroup Condition="$(Configuration.StartsWith('Linux'))">Linux</OSGroup>
+    <OSGroup Condition="$(Configuration.StartsWith('OSX'))">OSX</OSGroup>
+    <OSGroup Condition="$(Configuration.StartsWith('FreeBSD'))">FreeBSD</OSGroup>
+    <OSGroup Condition="$(Configuration.StartsWith('NetBSD'))">NetBSD</OSGroup>
   </PropertyGroup>
 
+  <!-- Setup Default symbol and optimization for Configuration -->
   <PropertyGroup>
-    <SignAssembly Condition="'$(UsingMicrosoftNETSdk)'!='true'">false</SignAssembly>
+    <!-- Ensure a portable PDB is emitted. A PDB is needed for crossgen. -->
+    <DebugType>Portable</DebugType>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(ConfigurationGroup)' == 'Debug'">
+    <Optimize Condition="'$(Optimize)' == ''">false</Optimize>
+    <DefineConstants>$(DefineConstants);DEBUG</DefineConstants>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(ConfigurationGroup)' == 'Checked'">
+    <Optimize Condition="'$(Optimize)' == ''">true</Optimize>
+    <DefineConstants>$(DefineConstants);DEBUG</DefineConstants>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(ConfigurationGroup)' == 'Release'">
+    <Optimize Condition="'$(Optimize)' == ''">true</Optimize>
+  </PropertyGroup>
+
+   <!-- Setup properties per OS -->
+  <Choose>
+    <When Condition="'$(OSGroup)' == 'AnyOS'">
+      <PropertyGroup>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(OSGroup)' == 'Windows_NT'">
+      <PropertyGroup>
+        <!-- Since cross compilation of test builds on Windows is possible, the
+             TargetsWindows property may already be set. Only set the property if
+             it is not already defined -->
+        <TargetsWindows Condition="'$(TargetsWindows)' == ''">true</TargetsWindows>
+        <TestNugetRuntimeId>win-$(ArchGroup)</TestNugetRuntimeId>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(OSGroup)' == 'Linux'">
+      <PropertyGroup>
+        <TargetsUnix>true</TargetsUnix>
+        <TargetsLinux>true</TargetsLinux>
+        <TestNugetRuntimeId>ubuntu.14.04-$(ArchGroup)</TestNugetRuntimeId>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(OSGroup)' == 'OSX'">
+      <PropertyGroup>
+        <TargetsUnix>true</TargetsUnix>
+        <TargetsOSX>true</TargetsOSX>
+        <TestNugetRuntimeId>osx.10.12-$(ArchGroup)</TestNugetRuntimeId>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(OSGroup)' == 'FreeBSD'">
+      <PropertyGroup>
+        <TargetsUnix>true</TargetsUnix>
```

Comment:
This is all duplicated now with libraries OSGroup logic, can't we share that?

Reply:
We will be able to share it with @Anipik's changes but it isn't possible today.